### PR TITLE
GEODE-8909: Support Redis HDEL command

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by sabbey37 — CHANGES_REQUESTED** [review]
Looks good! Just didn't think we should remove one of the test scenarios for `HEXISTS`.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java [hunk 1]
```diff
@@ -363,12 +384,6 @@ public void testHExists() {
     jedis.hset(key, field, value);
 
     assertThat(jedis.hexists(key, field)).isTrue();
-
-    jedis.hdel(key, field);
-
-    assertThat(jedis.hget(key, field)).isNull();
-    assertThat(jedis.hexists(key, field)).isFalse();
-
```
[the remark this answers is not in the document]
Comment (sabbey37) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java:371]: Actually, looking at the PR for `HEXISTS`, I see that these were separated out. So I'm ok with us deleting it in this PR.

[the remark this answers is not in the document]
Comment (jdeppe-pivotal) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java]: I've just rebased and this was already refactored into another test method.

**Comment by jdeppe-pivotal** [comment]
Also added a test for argument validation.

**Review by sabbey37 — APPROVED** [review]

**Pull request merged**